pkg/k8s/allowlist_test.go

```diff
@@ -15,35 +15,86 @@
 package k8s
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/crd/crdloader"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-func TestIsValueInAllowlist(t *testing.T) {
+func TestSupportsStateIntoSpecMerge(t *testing.T) {
 	tests := []struct {
 		name           string
-		kind           string
+		gvk            schema.GroupVersionKind
 		expectedResult bool
 	}{
 		{
-			name:           "ComputeInstance should support 'state-into-spec: merge'",
-			kind:           "ComputeInstance",
+			name: "ComputeInstance should support 'state-into-spec: merge'",
+			gvk: schema.GroupVersionKind{
+				Group:   "compute.cnrm.cloud.google.com",
+				Version: "v1beta1",
+				Kind:    "ComputeInstance",
+			},
 			expectedResult: true,
 		},
 		{
 			name: "AccessContextManagerServicePerimeterResource should not " +
 				"support 'state-into-spec: merge'",
-			kind:           "AccessContextManagerServicePerimeterResource",
+			gvk: schema.GroupVersionKind{
+				Group:   "accesscontextmanager.cnrm.cloud.google.com",
+				Version: "v1beta1",
+				Kind:    "AccessContextManagerServicePerimeterResource",
+			},
 			expectedResult: false,
 		},
 	}
 
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
-			actualResult := supportsStateIntoSpecMergeInKind(tc.kind)
+			actualResult := supportsStateIntoSpecMerge(tc.gvk)
 			if actualResult != tc.expectedResult {
 				t.Fatalf("got %v, want %v", actualResult, tc.expectedResult)
 			}
 		})
 	}
 }
+
+func TestOutputOnlyFieldsAreUnderObservedState(t *testing.T) {
```

Comment:
Hmmm, after some thinking, I propose this test should test "output-only fields are under status" situation.

CRD existing in the allowlist means that:
(1) currently all its output-only fields are directly under `status`,
(2) but it's still possible for it to have `status.observedState` with the observed fields configuration.
-> CRD existing in the allowlist should not have `status.observedState` unless it supports observed fields.

CRD not existing in the allowlist means that:
(1) currently all its output-only fields are under `status.observedState`,
(2) but it's still possible for it to not have any output-only fields, thus not have `status.observedState`.
-> It's valid for CRD not existing in the allowlist to have or not have `status.observedState`, and this is not easily detectable with our current code logic.